src/services/safes.rs

```diff
@@ -1,11 +1,119 @@
-use crate::models::service::safes::SafeInfoEx;
+use crate::cache::cache_operations::RequestCached;
+use crate::config::{base_transaction_service_url, transaction_request_timeout};
+use crate::models::backend::transactions::{MultisigTransaction, Transaction};
+use crate::models::backend::transfers::Transfer;
+use crate::models::commons::Page;
+use crate::models::service::safes::{SafeLastChanges, SafeState};
 use crate::providers::info::{DefaultInfoProvider, InfoProvider};
 use crate::utils::context::Context;
 use crate::utils::errors::ApiResult;
 
-pub fn get_safe_info_ex(context: &Context, safe_address: &String) -> ApiResult<SafeInfoEx> {
+pub fn get_safe_info_ex(context: &Context, safe_address: &String) -> ApiResult<SafeState> {
     let mut info_provider = DefaultInfoProvider::new(context);
     let safe_info = info_provider.safe_info(safe_address)?;
+    let safe_info_ex = safe_info.to_safe_info_ex(&mut info_provider);
 
-    Ok(safe_info.to_safe_info_ex(&mut info_provider))
+    let safe_state = SafeState {
+        safe_config: safe_info_ex,
+        safe_state: SafeLastChanges {
+            collectibles_tag: get_last_collectible(context, safe_address)
+                .unwrap_or(0)
+                .to_string(),
+            tx_queued_tag: get_last_queued_tx(context, safe_address)
+                .unwrap_or(0)
+                .to_string(),
+            tx_history_tag: get_last_history_tx(context, safe_address)
+                .unwrap_or(0)
+                .to_string(),
+        },
+    };
+
+    Ok(safe_state)
+}
+
+fn get_last_collectible(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let url = format!(
+        "{}/v1/safes/{}/transfers/?\
+        &ordering=executionDate\
+        &limit=1",
+        base_transaction_service_url(),
+        safe_address,
+    );
+
+    let body = RequestCached::new(url)
+        .request_timeout(transaction_request_timeout())
+        .execute(context.client(), context.cache())?;
+    let transaction: Page<Transfer> = serde_json::from_str(&body)?;
+
+    log::debug!("{:#?}", &transaction);
+    transaction
+        .results
+        .get(0)
+        .as_ref()
+        .map(|transfer| match transfer {
+            Transfer::Erc721(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Erc20(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Ether(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Unknown => 0,
+        })
+        .ok_or(api_error!("Couldn't get tx timestamps"))
+}
+
+fn get_last_queued_tx(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let url = format!(
+        "{}/v1/safes/{}/multisig-transactions/?\
+        &ordering=submissionDate\
```

Comment:
Was actually playing around with modified and it seems this is not updated when a new submission is added. I am not sure if we have a good way to detect a change in the queued without that 🤔 

How about we just set it to a random value and let it get invalidated via the cache by a hook xD ... idk ... I have no good idea right now.

Reply:
that would work, but we'd need a deterministic way of generating that, right? 

What if we replicate the same url for the `transactions/queued` endpoint? I got a bit confused by the `trusted` flag. Do the clients send it? Or is it for our debugging?

Reply:
> that would work, but we'd need a deterministic way of generating that, right?
Not really .. either that random value present in cache (as part of the cached response) and we return it or not then we create a new one and return it

> What if we replicate the same url for the transactions/queued endpoint? I got a bit confused by the trusted flag.
The problem is that we need to look at the whole response from the queued endpoint (which potentially are multiple pages)